senders/solomon/solomon_test.go

```diff
@@ -3,32 +3,186 @@ package solomon
 import (
 	"encoding/json"
 	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"runtime"
 	"testing"
 
 	"github.com/combaine/combaine/common/logger"
 	"github.com/combaine/combaine/common/tasks"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestSolomonSend(t *testing.T) {
+func TestNewWorker(t *testing.T) {
+	w := NewWorker(0, 0)
+	assert.Equal(t, 0, w.Id)
+	assert.Equal(t, 0, w.Retry)
+}
+
+func TestNewSolomonClient(t *testing.T) {
+	cli, _ := NewSolomonClient(SolomonCfg{Api: "api.addr"}, "id")
+	scl := cli.(*solomonClient)
+	assert.Equal(t, scl.id, "id")
+	assert.Equal(t, scl.Api, "api.addr")
+}
+
+func TestStartWorkers(t *testing.T) {
+	j := make(chan Job, 1)
+	StartWorkers(j)
+
+	j <- Job{PushData: []byte{}, SolCli: &solomonClient{SolomonCfg: SolomonCfg{Api: "bad://proto"}}}
+	for i := 3; i > 0; i-- {
+		if len(j) == 0 {
+			close(j)
+			break
+		}
+		runtime.Gosched()
+	}
+	assert.Equal(t, 0, len(j))
+}
+func TestRequest(t *testing.T) {
+	netAddr := "127.0.0.1:35313"
+
+	// for network timeout test
+	l, _ := net.Listen("tcp", netAddr)
```

Comment:
Its a blackhole for net.Dial.
With ctxhttp this test actually not need, but with httpclient it should cover case where net timeout ocured.